lib/tree/tree.go

```diff
@@ -0,0 +1,98 @@
+package tree
+
+import (
+	"github.com/dedis/cothority/lib/dbg"
+)
+
+/*
+The tree library holds all hosts and allows for different ways to
+retrieve the hosts - either in a n-ary tree, broadcast or just
+ a selection.
+*/
+
+// Tree holds the structure about the current tree
+type Tree struct {
```

Comment:
Are these structs intended to be protobuf-encodable?  If so, mention that in comments.  If not, what is the encoding plan for communicating trees?  And is it possible to communicate just part of a tree rather than the whole thing?  (Might not be critical from the start but worth thinking about.)